ftd/src/ast/component.rs

```diff
@@ -441,7 +449,55 @@ impl Loop {
 
         let loop_statement = loop_header.value.string(doc_id)?;
 
-        let (on, alias) = ftd::ast::utils::split_at(loop_statement.as_str(), ftd::ast::utils::AS);
+        let is_for_loop = loop_header.key.eq(ftd::ast::utils::FOR);
+
+        let (alias, on, key_name) = if is_for_loop {
+            let (pair, on) =
+                ftd::ast::utils::split_at(loop_statement.as_str(), ftd::ast::utils::IN);
+
+            let on = if let Some(on) = on {
+                on
+            } else {
+                return ftd::ast::parse_error(
+                    "Statement \"for\" needs a list to operate on",
+                    doc_id,
+                    loop_header.line_number,
+                );
+            };
+
+            let (alias, key_name) = ftd::ast::utils::split_at(pair.as_str(), ", ");
+
+            (alias, on, key_name)
+        } else {
+            use colored::Colorize;
+
+            println!(
+                "{}",
+                "Warning: \"$loop$\" is deprecated, use \"for:\" instead".bright_yellow()
```

Comment:
Also fix this warning message typo 
```suggestion
                "Warning: \"$loop$\" is deprecated, use \"for\" instead".bright_yellow()
```